h/static/scripts/directive/annotation.js

```diff
@@ -426,23 +445,26 @@ function AnnotationController(
             errorMessage(reason), 'Deleting annotation failed');
         };
         $scope.$apply(function() {
-          annotationMapper.deleteAnnotation(model).then(
+          annotationMapper.deleteAnnotation(domainModel).then(
             null, onRejected);
         });
       }
     }, true);
   };
 
+  /** Return some metadata extracted from the annotated document. */
+  vm.document = function() {
```

Comment:
Ok, but for now I prefer to remove data duplication, until we've got this code under control, then we can optimise as and when necessary. Good to know, though